tests/cases/conformance/types/typeParameters/typeParameterLists/typeParameterConstModifiersReverseMappedTypes.ts

```diff
@@ -0,0 +1,41 @@
+// @strict: true
+// @noEmit: true
+
+declare function test1<const T>(obj: {
```

Comment:
The first 3 test cases are testing objects and how `readonly`, `-readonly`, and lack of them impact the `T` and the fresh object that is created by mapping over `T`.

Those objects today would be constified for the most part but their properties wouldn't always be marked as `readonly` when they should be. When `readonly` modifier is used on the mapped type then that result (`typeof obj`) **has** `readonly` properties but the `const` type parameter itself doesn't have them. This is consistent with https://github.com/microsoft/TypeScript/pull/12589 . You might also want to recheck when/how `readonly` is preserved/stripped on members of the reverse mapped type [here](https://github.dev/microsoft/TypeScript/blob/2738fbf1e05b9429d03b5956ba0413d92f5b993c/src/compiler/checker.ts#L13491)